### DISCUSSION TITLE
How MenuBar works when it closed?

### DISCUSSION THREAD

**Discussion by lon9man** [NativePHP/laravel#213] [Q&A] [answered]
Hello, 
### What i did
1. app has 1 route with Livewire-component `Foo`
2. `Foo`-component on backend makes API request and saves the response into local file
3. `Foo`-component has attribute `wire:poll.10s="refresh"` to repeatedly call API on backend

**NOTE:**
i don't want to open any other window, because thought that app will work using `MenuBar` only.

### What i get
a) `MenuBar` with component is showed --> `refresh` **runs as expected**
b) `MenuBar` with component is closed --> `refresh` **DOESN'T run**

added `Window` into app with the same component `Foo` and minimized it after start.
results:
c) `Window` is showed      --> `refresh` **runs as expected**
d) `Window` is minimized --> `refresh` **runs as expected**

### What i expected
that my Livewire-component will work ONLY using `MenuBar` without any other minimized windows

**Comment by simonhamp** [comment, marked as answer]
It sounds like you're trying to have a persistent repeating task running _without requiring any windows to be open_.

For that, you shouldn't rely on the web view Javascript (which is what `wire:poll` uses), as it's only going to be available to execute if there's a window open on the page where `wire:poll` is running.

You should use the [Laravel Scheduler](https://laravel.com/docs/scheduling). [Here's an example](https://github.com/simonhamp/cronikl/blob/main/app/Console/Kernel.php#L23-L43) in a NativePHP app based on Laravel 10.

Hope that helps 😊
